### PR TITLE
chore: validate priority and weight ranges

### PR DESCRIPTION
**Pull request by hamidzr**
## Commentary (optional)

through jobs api

## Checklist

- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: master/internal/job/job.go [hunk 1]
```diff
@@ -178,18 +179,26 @@ func (j *Jobs) Receive(ctx *actor.Context) error {
...
+				if priority < 1 || priority > 99 {
+					errors = append(errors, "priority must be between 1 and 99")
+					continue
+				}
 				resp := ctx.Ask(jobActor, SetGroupPriority{
 					Priority: &priority,
 				})
 				if err := resp.Error(); err != nil {
-					errors = fmt.Sprintf("%s \n %s", errors, err.Error())
+					errors = append(errors, err.Error())
```
Comment (eecsliu) [inline master/internal/job/job.go:190]: Cool, so appending errors works?

**Review by eecsliu — APPROVED** [review]
Looking good!👌👌👌

File: master/internal/job/job.go (hunk 1 above)
Comment (hamidzr) [reply master/internal/job/job.go:190]: wdy mean? we still turn them into a single error before responding

**Comment by hamidzr** [comment]
a fix was added to master for the failing test. I'll rebase

**Pull request merged**